### PR TITLE
Use Cantera Error in leftover examples

### LINKED ISSUE #569: The examples should use except ct.CanteraError where appropriate
In some of the flame examples [here](https://github.com/Cantera/cantera/blob/5c783c708f0c08def6f6a3d8328b336d65842357/interfaces/cython/cantera/examples/reactors/ic_engine.py#L149), [here](https://github.com/Cantera/cantera/blob/5c783c708f0c08def6f6a3d8328b336d65842357/interfaces/cython/cantera/examples/onedim/diffusion_flame_extinction.py#L116), [here](https://github.com/Cantera/cantera/blob/5c783c708f0c08def6f6a3d8328b336d65842357/interfaces/cython/cantera/examples/onedim/diffusion_flame_batch.py#L119), and [here](https://github.com/Cantera/cantera/blob/5c783c708f0c08def6f6a3d8328b336d65842357/interfaces/cython/cantera/examples/onedim/diffusion_flame_batch.py#L170) we use a bare `except Exception`, probably as a leftover from before `CanteraError`s could be caught in Python. The latter two examples rely on catching a bare `Exception` to determine extinction, but it would probably be better to give a class name to that exception to distinguish solver errors vs. extinction.

### PR DESCRIPTION
**Pull request by CyberDrudge**
Fixes #569 
Implemented use of CanteraError in examples [diffusion_flame_extinction.py](https://github.com/Cantera/cantera/blob/master/interfaces/cython/cantera/examples/onedim/diffusion_flame_extinction.py) and [diffusion_flame_batch.py](https://github.com/Cantera/cantera/blob/master/interfaces/cython/cantera/examples/onedim/diffusion_flame_batch.py)

### CODE REVIEW

File: interfaces/cython/cantera/examples/onedim/diffusion_flame_batch.py [hunk 1]
```diff
@@ -76,8 +85,8 @@ def interrupt_extinction(t):
 
 # Exponents for the initial solution variation with changes in pressure Taken
 # from Fiala and Sattelmayer (2014). The exponents are adjusted such that the
-# strain rates increases proportional to p^(3/2), which results in flames
-# similar with respect to the extinction strain rate.
+# strain rates increases proportional to p^(3/2), which results in similar
+# flames with respect to the extinction strain rate.
```
Comment (bryanwweber) [inline interfaces/cython/cantera/examples/onedim/diffusion_flame_batch.py]: I think these lines should actually not be changed. In this case, the comment is using "similar" in a technical sense to mean that there is a similarity relationship between the conditions.

File: interfaces/cython/cantera/examples/onedim/diffusion_flame_extinction.py [hunk 2]
```diff
@@ -113,11 +113,11 @@
     f.set_profile('lambda', normalized_grid, f.L * strain_factor ** exp_lam_a)
     try:
         f.solve(loglevel=0)
-    except Exception as e:
-        # Throw Exception if solution fails
+    except ct.CanteraError as e:
+        # Throw Error if solution fails
```
Comment (bryanwweber) [inline interfaces/cython/cantera/examples/onedim/diffusion_flame_extinction.py]: I think you can probably just delete the comment, it doesn't add anything anyways.

**Review by bryanwweber — CHANGES_REQUESTED** [review]
Thanks @CyberDrudge I restarted the failing build because I think it was a problem with the CodeCov service, not with anything you did here. A few changes below!

**Comment by CyberDrudge** [comment]
I applied the suggested changes and squashed the commits.

**Review by bryanwweber — APPROVED** [review]

**Pull request merged**